joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/astcreation/AstForDeclarationsCreator.scala

```diff
@@ -23,8 +23,10 @@ trait AstForDeclarationsCreator(implicit withSchemaValidation: ValidationMode) {
       .filename(relativeFileName)
       .fullName(fullName)
     methodAstParentStack.push(namespaceBlock)
+    scope.pushNewScope(namespaceBlock)
```

Comment:
A note for myself, but I think we should eventually get rid of `methodAstParentStack` if we are going to use `scope`. I see `jssrc2cpg` uses a custom `Scope` object, but we could extend it to add some more relevant features.